feature_engine/selection/drop_psi_features.py

```diff
@@ -0,0 +1,566 @@
+        self.variables = _check_input_parameter_variables(variables)
+
+    def fit(self, X: pd.DataFrame, y: pd.Series = None):
+        """
+        Assess the features that needs to be dropped because of high PSI values.
+
+        Parameters
+        ----------
+        X : pandas dataframe of shape = [n_samples, n_features]
+            The training dataset.
+
+        y : pandas series. Default = None
+            y is not needed in this transformer. You can pass y or None.
+
+        Returns
+        -------
+        self
+        """
+        # check input dataframe
+        X = _is_dataframe(X)
+
+        # find all variables or check those entered are present in the dataframe
+        self.variables_ = _find_or_check_numerical_variables(X, self.variables)
+
+        # Remove the split columns from the variables list. It is automatically added to
+        # it in variables is not defined at initiation.
+        if self.split_col in self.variables_:
+            self.variables_.remove(self.split_col)
+
+        if self.missing_values == "raise":
+            # check if dataset contains na
+            _check_contains_na(X, self.variables_)
+            _check_contains_inf(X, self.variables_)
+
+        # Split the dataframe into a basis and a measurement dataframe.
+        basis_df, measurement_df = self._split_dataframe(X)
+
+        # Switch base and measurement dataframe if required.
+        if self.switch:
+            measurement_df, basis_df = basis_df, measurement_df
+
+        # Compute the PSI
+        self.psi_ = self._compute_PSI(basis_df, measurement_df, self.bucketer)
+
+        # Select features below the threshold
+        self.features_to_drop_ = self.psi_[
+            self.psi_.value >= self.threshold
+        ].index.to_list()
+
+        self.n_features_in_ = X.shape[1]
+
+        return self
+
+    def _compute_PSI(self, df_basis, df_meas, bucketer):
+        """
+        Compute the PSI by first bucketing (=binning) and then call the psi calculator
+        for each feature.
+
+        Parameters
+        ----------
+        df_basis : pandas dataframe of shape = [m_rows, n_features]
+            Serve as reference for the features distribution.
+
+        df_meas : pandas dataframe of shape = [p_rows, n_features]
+            Matrix for which the feature distributions will be compared to the basis.
+
+        bucketer : EqualFrequencyDiscretiser or EqualWidthDiscretiser.
+                Default = EqualFrequencyDiscretiser
+            Class used to bucket (bin) the features in order to approximate
+            the distribution.
+
+        Returns
+        -------
+        results_df: pandas dataframe.
+            Dataframe containing the PSI for each feature.
+        """
+        # Initialize a container for the results.
+        results = {}
+
+        # Compute the PSI for each feature excluding the column used for split.
+        for feature in self.variables_:
```

Comment:
with the code as it is, we are training 1 discretizer per variable, when we could apply the discretizer to the entire dataframe. I would change

Reply:
We could if there are no missing values in the dataframe. Otherwise we need to either use .dropna() or impute the missings. In the two cases we change the distribution of the features which then affects the values of the PSI.
In my first draft I was using the discretizer on the full dataframe but I changed it back to train a single discretizer per variable... maybe there is a way around this but I haven't find it (yet).

In banking industrie, the number of missing values can be very large. This happens for example looking at customer's profiles. Most customers have a current account but only a small part of the customers has products like  personal loans or securities. The feature with the total balance / debt is missing for everyone that does not have the product.